src/video.h

```diff
@@ -11,11 +11,183 @@
+
+    encoder_platform_formats_avcodec(
+      const AVHWDeviceType &avcodec_base_dev_type,
+      const AVHWDeviceType &avcodec_derived_dev_type,
+      const AVPixelFormat &avcodec_dev_pix_fmt,
+      const AVPixelFormat &avcodec_pix_fmt_8bit,
+      const AVPixelFormat &avcodec_pix_fmt_10bit,
+      const init_buffer_function_t &init_avcodec_hardware_input_buffer_function):
+        avcodec_base_dev_type { avcodec_base_dev_type },
+        avcodec_derived_dev_type { avcodec_derived_dev_type },
+        avcodec_dev_pix_fmt { avcodec_dev_pix_fmt },
+        avcodec_pix_fmt_8bit { avcodec_pix_fmt_8bit },
+        avcodec_pix_fmt_10bit { avcodec_pix_fmt_10bit },
+        init_avcodec_hardware_input_buffer { init_avcodec_hardware_input_buffer_function } {
+      dev_type = map_base_dev_type(avcodec_base_dev_type);
+      pix_fmt_8bit = map_pix_fmt(avcodec_pix_fmt_8bit);
+      pix_fmt_10bit = map_pix_fmt(avcodec_pix_fmt_10bit);
+    }
+
+    AVHWDeviceType avcodec_base_dev_type, avcodec_derived_dev_type;
+    AVPixelFormat avcodec_dev_pix_fmt;
+    AVPixelFormat avcodec_pix_fmt_8bit, avcodec_pix_fmt_10bit;
+
+    init_buffer_function_t init_avcodec_hardware_input_buffer;
+  };
+
+  struct encoder_platform_formats_nvenc: encoder_platform_formats_t {
+    encoder_platform_formats_nvenc(
+      const platf::mem_type_e &dev_type,
+      const platf::pix_fmt_e &pix_fmt_8bit,
+      const platf::pix_fmt_e &pix_fmt_10bit) {
+      encoder_platform_formats_t::dev_type = dev_type;
+      encoder_platform_formats_t::pix_fmt_8bit = pix_fmt_8bit;
+      encoder_platform_formats_t::pix_fmt_10bit = pix_fmt_10bit;
+    }
+  };
+
+  struct encoder_t {
+    std::string_view name;
+    enum flag_e {
+      PASSED,  // Is supported
+      REF_FRAMES_RESTRICT,  // Set maximum reference frames
+      CBR,  // Some encoders don't support CBR, if not supported --> attempt constant quantatication parameter instead
+      DYNAMIC_RANGE,  // hdr
+      VUI_PARAMETERS,  // AMD encoder with VAAPI doesn't add VUI parameters to SPS
+      MAX_FLAGS
+    };
+
+    static std::string_view
+    from_flag(flag_e flag) {
+#define _CONVERT(x) \
+  case flag_e::x:   \
+    return #x##sv
+      switch (flag) {
+        _CONVERT(PASSED);
+        _CONVERT(REF_FRAMES_RESTRICT);
+        _CONVERT(CBR);
+        _CONVERT(DYNAMIC_RANGE);
+        _CONVERT(VUI_PARAMETERS);
+        _CONVERT(MAX_FLAGS);
+      }
+#undef _CONVERT
+
+      return "unknown"sv;
+    }
+
+    struct option_t {
+      KITTY_DEFAULT_CONSTR_MOVE(option_t)
+      option_t(const option_t &) = default;
+
+      std::string name;
+      std::variant<int, int *, std::optional<int> *, std::function<int()>, std::string, std::string *> value;
+
+      option_t(std::string &&name, decltype(value) &&value):
+          name { std::move(name) }, value { std::move(value) } {}
+    };
+
+    const std::unique_ptr<const encoder_platform_formats_t> platform_formats;
+
+    struct profile {
```

Comment:
Note, this was `struct {` before. Changing this allowed the docs to build without the `Duplicate C++ Declaration` warning.

Reply:
It's good practice to give all types --- especially complex ones --- clear names. And if it was hard to name, that's a code smell that it doesn't have a cohesive, single responsibility.